src/bootstrap/src/core/sanity.rs

```diff
@@ -177,6 +177,41 @@ than building it.
             continue;
         }
 
+        // Check if there exists a built-in target in the list of supported targets.
+        let cur_dir = env::current_dir().unwrap().display().to_string();
+        
+        let supported_target_file = format!("{cur_dir}/compiler/rustc_target/src/spec/mod.rs");
+        let contents = fs::read_to_string(supported_target_file)
+            .expect("Should have been able to read the file.");
```

Comment:
This approach is too fragile I think. I suggest moving the target list to a separate text file within the "/compiler/rustc_target/src/spec" folder. Then use [include_str!](https://doc.rust-lang.org/std/macro.include_str.html) to access them from both compiler and bootstrap sources.

Reply:
That sounds dubious too. Depending on the goal here, I would consider using `rustc --print target-list` or `rustc --print all-target-specs-json` if we need more information on each target. That *does* have the limitation that we're only going to see stage0 targets (anything further won't be available at this point in bootstrap), but it feels similar to the problem we already have of adding to the check-cfg list (https://github.com/rust-lang/rust/blob/01f7f3a1ff983ddfe485bbd9cedd59c378fdaceb/src/bootstrap/src/lib.rs#L94-L97) so I don't see any problem maintaining a similar list of recently added target triples.

Reply:
Yeah, that seems quite better.